crates/forge/src/collecting.rs

```diff
@@ -0,0 +1,588 @@
+            .collect()
+    } else {
+        tests_from_crates
+    }
+}
+
+pub fn pack_tests_into_one_file(
+    tmp_dir: &TempDir,
+    tests_folder_path: &Utf8PathBuf,
+) -> Result<TestCrate> {
+    tmp_dir
+        .copy_from(tests_folder_path, &["**/*.cairo"])
+        .context("Unable to copy files to temporary directory")?;
+
+    let tests_lib_path = tmp_dir.child("lib.cairo");
+    assert!(
+        !(tests_lib_path.try_exists()?),
+        "Path = {:?} already exists",
+        tests_lib_path.path()
+    );
+    tests_lib_path.touch()?;
+
+    let mut content = String::new();
+    for entry in WalkDir::new(tests_folder_path)
+        .max_depth(1)
+        .sort_by_file_name()
+    {
+        let entry = entry
+            .with_context(|| format!("Failed to read directory at path = {tests_folder_path}"))?;
+        let path = entry.path();
+
+        if path.is_file() && path.extension().unwrap_or_default() == "cairo" {
+            let mod_name = path
+                .strip_prefix(tests_folder_path)
+                .expect("Each test file path should start with package path")
+                .to_str()
+                .context("Unable to convert test file path to string")?
+                .strip_suffix(".cairo")
+                .expect("Each test file path should have .cairo extension");
+
+            content.push_str(&format!("mod {mod_name};\n"));
+        }
+    }
+
+    std::fs::write(tests_lib_path, content).context("Failed to write to tests lib file")?;
+
+    let tests_tmp_dir_path = Utf8PathBuf::from_path_buf(tmp_dir.to_path_buf())
+        .map_err(|_| anyhow!("Failed to convert tests temporary directory to Utf8PathBuf"))?;
+
+    Ok(TestCrate {
+        crate_root: tests_tmp_dir_path,
+        crate_name: "tests".to_string(),
+        crate_type: TestCrateType::Tests,
+    })
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::scarb::ForgeConfig;
+    use assert_fs::fixture::PathCopy;
+    use assert_fs::TempDir;
+    use test_collector::ExpectedTestResult;
+
+    #[test]
+    fn runner_config_argument_precedence() {
+        let config_from_scarb = ForgeConfig {
+            exit_first: false,
+            fork: vec![],
+            fuzzer_runs: Some(1234),
+            fuzzer_seed: Some(1000),
+        };
+        let config = RunnerConfig::new(
+            Default::default(),
+            None,
+            false,
+            true,
+            Some(100),
+            Some(32),
+            &config_from_scarb,
+        );
+        assert_eq!(
+            config,
+            RunnerConfig {
+                workspace_root: Default::default(),
+                test_name_filter: None,
+                exact_match: false,
+                exit_first: true,
+                fork_targets: vec![],
+                fuzzer_runs: 100,
+                fuzzer_seed: 32,
+            }
+        );
+    }
+
+    #[test]
+    fn collecting_test_crates() {
```

Comment:
collecting_test_compilation_targets

Reply:
Test cases:
How it behaves when there is no lib
How it behaves where there are no tests only lib

Reply:
> How it behaves where there are no tests only lib

I don't think we need to test that, it is kinda testing the behaviour of the compiler

> How it behaves when there is no lib

It is already tested in e2e tests: `collection_with_lib` and `collection_without_lib` packages

Reply:
> Test cases: How it behaves when there is no lib 
> How it behaves where there are no tests only lib

so in both these cases the returned struct will contain a `TestCompilationTarget` representing a lib. It's hardcoded in the assumption that this always exists. 
The path to the lib is passed by the user as an argument, so I'm unsure if we should verify it's existence inside this method.

Reply:
Still imo e2e tests are enough there

Reply:
> >   How it behaves where there are no tests only lib
> 
> I don't think we need to test that, it is kinda testing the behaviour of the compiler

I am not convinced, compiler doesn't know about test dir, we implement that on the top of it. How collection is going to behave if we call it on a dir without such directory?